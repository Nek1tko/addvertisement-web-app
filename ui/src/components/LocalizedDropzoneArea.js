import { DropzoneArea } from 'material-ui-dropzone';

const getFileLimitExceedMessage = (filesLimit) => {
    return "Достигнуто максимальное количетсов файлов: " + filesLimit;
};

const getFileAddedMessage = (fileName) => {
    return "Файл " + fileName + " успешно загружен"
};

const getFileRemovedMessage = (fileName) => {
    return "Файл " + fileName + " удален";
};

const getDropRejectMessage = (
    rejectedFile,
    acceptedFiles,
    maxFileSize
) => {
    return "Файл " + rejectedFile.name + " отклонен. Файл не поддерживается или размер файла больше "
        + (maxFileSize / 1024 ** 2) + " МБ";
};


export default function LocalizedDropzoneArea(props) {
    return (
        <DropzoneArea
            filesLimit={props.filesLimit}
            maxFileSize={props.maxFileSize} // 3 mb
            acceptedFiles={props.acceptedFiles}
            onChange={props.onChange}
            dropzoneText="Перетащите сюда картинку или нажмите"
            getFileLimitExceedMessage={getFileLimitExceedMessage}
            getFileAddedMessage={getFileAddedMessage}
            getFileRemovedMessage={getFileRemovedMessage}
            getDropRejectMessage={getDropRejectMessage}
        >
        </DropzoneArea>
    );
}
