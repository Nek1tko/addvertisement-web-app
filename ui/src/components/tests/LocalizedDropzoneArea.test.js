import '../LocalizedDropzoneArea'
import LocalizedDropzoneArea from "../LocalizedDropzoneArea";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import {configure, mount, shallow} from 'enzyme';
import {DropzoneArea} from "material-ui-dropzone";

const props = {
    filesLimit: 3,
    maxFileSize: 3145728,
    acceptedFiles: [".jpeg", ".jpg"],
    onChange: jest.fn()
};

configure({adapter: new Adapter()});


describe('LocalizedDropZoneAreaTests', () => {
    test('LocalizedDropZoneAreaRenderTest', () => {
        mount(<LocalizedDropzoneArea{...props} />);
    });

    test('LocalizedDropZoneAreaSnapshotTest', () => {
        const wrapper = shallow(<LocalizedDropzoneArea{...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('LocalizedDropZoneAreaOnChangeTest', () => {
        const wrapper = shallow(<LocalizedDropzoneArea{...props} />);
        // find the DropzoneArea node
        const dropzoneAreaWrapper = wrapper.find(DropzoneArea);
        // call its onChange prop
        const onChange = 'onChange';
        dropzoneAreaWrapper.prop(onChange)();
        // check handleDropzoneChange has been called
        expect(props.onChange).toHaveBeenCalled();

        // check getFileAddedMessage function
        const fileName = 'file.jpg';
        const message = "Файл " + fileName + " успешно загружен";
        expect(dropzoneAreaWrapper.prop('getFileAddedMessage')(fileName)).toBe(message);
    });
});
