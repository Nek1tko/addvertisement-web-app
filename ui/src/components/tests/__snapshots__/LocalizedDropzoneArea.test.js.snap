// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalizedDropZoneAreaTests LocalizedDropZoneAreaSnapshotTest 1`] = `ShallowWrapper {}`;
